master/buildbot/reporters/stash.py

```diff
@@ -61,8 +67,22 @@ def send(self, build):
         else:
             status = STASH_INPROGRESS
             description = self.startDescription
+
+        # got_revision could be a string, a dictionary or None
+        got_revision = props.getProperty('got_revision', None)
         for sourcestamp in build['buildset']['sourcestamps']:
             sha = sourcestamp['revision']
+
+            if sha is None:
+                if isinstance(got_revision, dict):
```

Comment:
In `handle_pullrequest` [line 130](https://github.com/buildbot/buildbot/pull/3314/files#diff-6a2f55ab8b3902c7f4433724cbce39ffR130) revision is explicitly set to `None` (meaning the latest revision), I thought this way is cleaner than calling Bitbucket server API or running git from the buildbot master server